src/EventStore.Core/ClusterVNode.cs

```diff
@@ -1258,6 +1258,16 @@ public class ClusterVNode<TStreamId> :
 
 			// STORAGE SCAVENGER
 			ScavengerFactory scavengerFactory;
+			var scavengerDispatcher = new IODispatcher(_mainQueue, new PublishEnvelope(MainQueue));
```

Comment:
oh good catch - could just be worth checking if we need all of these subscriptions, although.. if its not obvious maybe err on the side of keeping them

Reply:
It looks like the scavenging IODispatcher only ever reads backwards or writes events, so I've removed the forward reads